extmod/moductypes.c

```diff
@@ -602,7 +602,7 @@ static mp_obj_t uctypes_struct_unary_op(mp_unary_op_t op, mp_obj_t self_in) {
                 uint agg_type = GET_TYPE(offset, AGG_TYPE_BITS);
                 if (agg_type == PTR) {
                     byte *p = *(void **)self->addr;
-                    return mp_obj_new_int((mp_int_t)(uintptr_t)p);
+                    return mp_obj_new_int_from_uint((mp_int_t)(uintptr_t)p);
```

Comment:
```suggestion
                    return mp_obj_new_int_from_uint((uintptr_t)p);
```

(Takes out the unnecessary and maybe-UB-probably-not cast.)

I ran into this testing #16311 V2 changes on qemu where the RAM range is at the top of the 32-bit space, I'm guessing you probably hit the same. :grin:

Reply:
Yes I hit this on qemu as well, testing #8381, which checks the address of ROMFS files are within the address of the ROMFS itself.